core/src/main/scala/kafka/raft/KafkaNetworkChannel.scala

```diff
@@ -115,7 +116,7 @@ class KafkaNetworkChannel(time: Time,
           if (client.connectionFailed(node)) {
             pendingOutbound.poll()
             val apiKey = ApiKeys.forId(request.data.apiKey)
-            val disconnectResponse = RaftUtil.errorResponse(apiKey, Errors.BROKER_NOT_AVAILABLE)
+            val disconnectResponse = RaftUtil.errorResponse(metadataTopicPartition, apiKey, Errors.BROKER_NOT_AVAILABLE)
```

Comment:
This exposes an issue I have been thinking about. Should we have a top-level error code for these APIs? Otherwise we are left with these awkward cases where we have to fill in a topic partition to ensure we have some place for the error code.